### PR TITLE
Support joins using a different column name

### LINKED ISSUE #152: Support Joins where column name is different from field
Currently, we can use Join to chain different searches using a common column name.

It would be nice to extend this to be able to Join a dataframe using a specified column name, that way we would be able to join f.ex. GrondwaterFilter with Boring.

F.ex.:
```
fs = GrondwaterFilterSearch()
bs = BoringSearch()

df_filters = fs.search(location=Box()), return_fields=('pkey_filter', 'boringfiche')
df_boringen = bs.search(query=Join(df_filters, on='pkey_boring', using='boringfiche'))
```

### PR DESCRIPTION
**Pull request by Roel**
* [x] I have read and followed the guidelines in the `CONTRIBUTING` document
* [x] I have checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change
* [x] I have updated the documentation accordingly.
* [x] I have added tests to cover my changes.
* [x] All new and existing tests passed.

Support Join queries where the column name in the dataframe is different from the one in the type being searched.

Closes #152

### CODE REVIEW

File: docs/query_attribute.rst [hunk 1]
```diff
@@ -330,3 +337,20 @@ The following example returns all the lithological descriptions of boreholes tha
...
+    from pydov.util.query import Join
+
+    from pydov.search.boring import BoringSearch
+    from pydov.search.grondwaterfilter import GrondwaterFilterSearch
+
+    fs = GrondwaterFilterSearch()
+    bs = BoringSearch()
+
+    filters = fs.search(location=WithinDistance(Point(96540, 186900), 10, 'km'),
+                        return_fields=('pkey_filter', 'boringfiche'))
```
Comment (stijnvanhoey) [inline docs/query_attribute.rst]: Could we illustrate the usage by providing an example of the top rows of the `filters` dataframe?

File: pydov/util/query.py [hunk 2]
```diff
@@ -62,34 +62,43 @@ class Join(PropertyInList):
...
+            Name of the column in the queried datatype to join on.
+        using : str, optional
+            Name of the column in the dataframe to use for joining. By
+            default, the same column name as in `on` is assumed.
 
         Raises
         ------
         ValueError
-            If the join_column is not present in the dataframe.
+            If the `using` column is not present in the dataframe.
```
Comment (stijnvanhoey) [inline pydov/util/query.py]: specify which dataframe: `not present in the ........ dataframe.`

File: pydov/util/query.py [hunk 3]
```diff
@@ -62,34 +62,43 @@ class Join(PropertyInList):
...
+
+            If `using` is None and the `on` column is not present in the
+            dataframe.
```
Comment (stijnvanhoey) [inline pydov/util/query.py]: specify which dataframe: `not present in the ........ dataframe.`

File: docs/query_attribute.rst (hunk 1 above)
Comment (Roel) [reply docs/query_attribute.rst]: Good idea, I added the top rows from both dataframes.

File: pydov/util/query.py (hunk 2 above)
Comment (Roel) [reply pydov/util/query.py]: There is only one dataframe: the one that is given in the `dataframe` parameter. The Join is a query builder that builds a query from that dataframe (which is in fact a list of PropertyIsEqualTo inside an Or). It is only when that Join query is used in a .search() of another type that the query (and thus the join) is made.

This also why we cannot validate the `on` column when creating a Join: at this point we don't know yet which datatype will be joined with.

Comment (stijnvanhoey) [reply pydov/util/query.py]: ok, that clarifies a lot.

**Pull request merged**